<a href="https://colab.research.google.com/github/Shambhuraje1919/Machine_Learning_Paractice/blob/main/PhoneRanking.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("shambhurajejagadale/mobile-phone-dataset-with-and-ranking")

print("Path to dataset files:", path)

100%|██████████| 655/655 [00:00<00:00, 1.19MB/s]

Extracting files...
Path to dataset files: /root/.cache/kagglehub/datasets/shambhurajejagadale/mobile-phone-dataset-with-and-ranking/versions/1


In [19]:
import os
import pandas as pd

In [20]:
os.listdir(path)

['final_dataset.csv']

In [23]:
df = pd.read_csv(os.path.join(path,"final_dataset.csv"))

In [24]:
df.head()

,Brand,Model,Price_USD,RAM_GB,Storage_GB,Battery_mAh,Screen_Size_Inch,Camera_MP,OS,Release_Year,Overall_Score,Rank
0,Samsung,Galaxy S21,799,8,128,4000,6.20,64,Android,2021,0.627,12
1,Apple,iPhone 13,999,4,128,3240,6.10,12,iOS,2021,0.354,13
2,Xiaomi,Redmi Note 11,199,6,128,5000,6.43,50,Android,2022,0.720,3
3,Realme,Realme 9 Pro,299,8,128,5000,6.60,64,Android,2022,0.776,2
4,OnePlus,OnePlus 9,729,8,128,4500,6.55,48,Android,2021,0.636,9


In [35]:
df.isnull().sum()

,0
Brand,0
Model,0
Price_USD,0
RAM_GB,0
Storage_GB,0
Battery_mAh,0
Screen_Size_Inch,0
Camera_MP,0
OS,0
Release_Year,0


::

,Price_USD,RAM_GB,Storage_GB,Battery_mAh,Screen_Size_Inch,Camera_MP,Release_Year,Overall_Score,Rank,Brand_Huawei,...,Model_OnePlus 9,Model_Oppo A54,Model_P40 Lite,Model_Poco X3 Pro,Model_Realme 9 Pro,Model_Redmi Note 11,Model_Vivo V21,Model_Zero X Pro,Model_iPhone 12,Model_iPhone 13
0,799,8,128,4000,6.20,64,2021,0.627,12,False,...,False,False,False,False,False,False,False,False,False,False
1,999,4,128,3240,6.10,12,2021,0.354,13,False,...,False,False,False,False,False,False,False,False,False,True
2,199,6,128,5000,6.43,50,2022,0.720,3,False,...,False,False,False,False,False,True,False,False,False,False
3,299,8,128,5000,6.60,64,2022,0.776,2,False,...,False,False,False,False,True,False,False,False,False,False
4,729,8,128,4500,6.55,48,2021,0.636,9,False,...,True,False,False,False,False,False,False,False,False,False
5,249,6,128,5000,6.51,13,2021,0.641,8,False,...,False,True,False,False,False,False,False,False,False,False
6,399,8,128,4000,6.44,64,2021,0.707,5,False,...,False,False,False,False,False,False,True,False,False,False
7,349,6,128,4200,6.40,48,2020,0.648,7,True,...,False,False,True,False,False,False,False,False,False,False
8,199,4,64,5050,6.52,48,2021,0.631,11,False,...,False,False,False,False,False,False,False,False,False,False
9,299,8,128,4500,6.67,108,2021,0.833,1,False,...,False,False,False,False,False,False,False,True,False,False


In [48]:
df.columns

Index(['Brand', 'Model', 'Price_USD', 'RAM_GB', 'Storage_GB', 'Battery_mAh',
       'Screen_Size_Inch', 'Camera_MP', 'OS', 'Release_Year', 'Overall_Score',
       'Rank'],
      dtype='object')

In [67]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split

In [ ]:
df =pd.get_dummies(df,columns=["Brand","OS","Model"],drop_first=True)

In [69]:
x = df.drop("Rank",axis=1)
y= df["Rank"]

In [85]:
x_train , x_test ,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=44)

In [86]:
model = RandomForestRegressor()

In [73]:
df.shape

(15, 34)

In [91]:
model.fit(x_train,y_train)

RandomForestRegressor()

In [92]:
pred = model.predict(x_test)

In [93]:
print(pred)
print(y_test)

[7.78 9.07 4.47]
5     8
8    11
6     5
Name: Rank, dtype: int64


In [94]:
from sklearn.metrics import accuracy_score , confusion_matrix

In [95]:
from sklearn.metrics import r2_score
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import mean_squared_error

print("R2:", r2_score(y_test, pred))
print("MAE:", mean_absolute_error(y_test, pred))
print("RMSE:", mean_squared_error(y_test, pred)**0.5)

R2: 0.7747666666666667
MAE: 0.8933333333333332
RMSE: 1.1624973118248487


In [83]:
df

,Price_USD,RAM_GB,Storage_GB,Battery_mAh,Screen_Size_Inch,Camera_MP,Release_Year,Overall_Score,Rank,Brand_Huawei,...,Model_OnePlus 9,Model_Oppo A54,Model_P40 Lite,Model_Poco X3 Pro,Model_Realme 9 Pro,Model_Redmi Note 11,Model_Vivo V21,Model_Zero X Pro,Model_iPhone 12,Model_iPhone 13
0,799,8,128,4000,6.20,64,2021,0.627,12,False,...,False,False,False,False,False,False,False,False,False,False
1,999,4,128,3240,6.10,12,2021,0.354,13,False,...,False,False,False,False,False,False,False,False,False,True
2,199,6,128,5000,6.43,50,2022,0.720,3,False,...,False,False,False,False,False,True,False,False,False,False
3,299,8,128,5000,6.60,64,2022,0.776,2,False,...,False,False,False,False,True,False,False,False,False,False
4,729,8,128,4500,6.55,48,2021,0.636,9,False,...,True,False,False,False,False,False,False,False,False,False
5,249,6,128,5000,6.51,13,2021,0.641,8,False,...,False,True,False,False,False,False,False,False,False,False
6,399,8,128,4000,6.44,64,2021,0.707,5,False,...,False,False,False,False,False,False,True,False,False,False
7,349,6,128,4200,6.40,48,2020,0.648,7,True,...,False,False,True,False,False,False,False,False,False,False
8,199,4,64,5050,6.52,48,2021,0.631,11,False,...,False,False,False,False,False,False,False,False,False,False
9,299,8,128,4500,6.67,108,2021,0.833,1,False,...,False,False,False,False,False,False,False,True,False,False


*The model:::: achieved an R² score of **0.775:\**, indicating good predictive performance on the available data. However, the dataset contains only 15 samples, making the evaluation less reliable and prone to variance. A larger dataset would be required to validate the model's real-world performance.*